### ISSUE TITLE
Python 3.9: ast.Param has been removed

### ISSUE THREAD

**Issue by vstinner** [malthe/chameleon#303]
Python 3.9 dropped ast.Param: https://bugs.python.org/issue39969

Full traceback of pyperformance benchmark:

```
Traceback (most recent call last):

  File "/home/travis/build/python/pyperformance/pyperformance/benchmarks/bm_chameleon.py", line 35, in <module>

    main()

  File "/home/travis/build/python/pyperformance/pyperformance/benchmarks/bm_chameleon.py", line 25, in main

    tmpl = PageTemplate(BIGTABLE_ZPT)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/zpt/template.py", line 200, in __init__

    super(PageTemplate, self).__init__(body, **config)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/template.py", line 132, in __init__

    self.write(body)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/template.py", line 227, in write

    self.cook(body)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/template.py", line 162, in cook

    program = self._cook(body, digest, names)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/template.py", line 237, in _cook

    source = self._compile(body, builtins)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/template.py", line 271, in _compile

    compiler = Compiler(

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 950, in __init__

    module.body += self.visit(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 979, in visit

    iterator = visitor(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1019, in visit_Module

    program = self.visit(node.program)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 979, in visit

    iterator = visitor(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1038, in visit_MacroProgram

    stmts = self.visit(macro)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 980, in visit

    return list(iterator)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1079, in visit_Macro

    nodes = itertools.chain(*tuple(map(self.visit, node.body)))

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 979, in visit

    iterator = visitor(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1054, in visit_Context

    self.visit(node.node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 980, in visit

    return list(iterator)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 984, in visit_Sequence

    for stmt in self.visit(item):

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 980, in visit

    return list(iterator)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1251, in visit_Define

    for stmt in self.visit(assignment):

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 979, in visit

    iterator = visitor(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 1206, in visit_Alias

    return self._engine(node.expression, target)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/compiler.py", line 776, in __call__

    self.visitor(stmt)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/astutil.py", line 982, in __call__

    self.visit(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/astutil.py", line 966, in visit

    super(AnnotationAwareVisitor, self).visit(node)

  File "/opt/python/3.9-dev/lib/python3.9/ast.py", line 395, in visit

    return visitor(node)

  File "/opt/python/3.9-dev/lib/python3.9/ast.py", line 403, in generic_visit

    self.visit(item)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/astutil.py", line 966, in visit

    super(AnnotationAwareVisitor, self).visit(node)

  File "/opt/python/3.9-dev/lib/python3.9/ast.py", line 395, in visit

    return visitor(node)

  File "/tmp/tmp9g0mj8lh/lib/python3.9/site-packages/chameleon/astutil.py", line 991, in visit_Name

    if isinstance(node.ctx, ast.Param):

AttributeError: module 'ast' has no attribute 'Param'
```

**Comment by isidentical** [comment]
python/cpython#19056

**Comment by malthe** [comment]
Something to keep an eye on, but the wind seems to blow in the direction of keeping backwards compatibility.

**Comment by dataflake** [comment]
The referenced PR has been merged into Python 3.9 before 3.9.0 came out, so this is no longer valid.
